src/Symfony/Component/HttpKernel/Event/GetResponseForExceptionEvent.php

```diff
@@ -19,45 +19,58 @@
  */
 class GetResponseForExceptionEvent extends RequestEvent
 {
-    /**
-     * The exception object.
-     *
-     * @var \Exception
-     */
-    private $exception;
+    private $throwable;
 
     /**
      * @var bool
      */
     private $allowCustomResponseCode = false;
 
-    public function __construct(HttpKernelInterface $kernel, Request $request, int $requestType, \Exception $e)
+    public function __construct(HttpKernelInterface $kernel, Request $request, int $requestType, $e)
     {
         parent::__construct($kernel, $request, $requestType);
 
-        $this->setException($e);
+        $this->throwable = $e;
     }
 
     /**
-     * Returns the thrown exception.
+     * Returns the thrown exception / error.
+     *
+     * @deprecated since Symfony 4.4, use "getThrowable()" instead
      *
-     * @return \Exception The thrown exception
+     * @return \Throwable The thrown exception / error
```

Comment:
Isn't it a BC break?

Reply:
In theory it could be a bc break for someone only accepting an Exception. But I don't see an alternative. This is the main thing the PR tries to achive: being able to access the raw Throwable without having to transform Error to Exception.

Reply:
I forgot this one as a BC break. I think `FatalThrowableError` is gonna stay in 4.4. This PR can aim at reducing its usage at the strict minimum and at deprecating the last places it is used (ie to accept \Throwable)